pkg/morph/client/client.go

```diff
@@ -204,6 +205,49 @@ func (c *Client) TestInvoke(contract util.Uint160, method string, args ...any) (
 	return val.Stack, nil
 }
 
+// TestInvokeIterator is the same [Client.TestInvoke] but expands an iterator placed
+// on the stack. Returned values are the values an iterator provides.
+func (c *Client) TestInvokeIterator(contract util.Uint160, method string, args ...any) (res []stackitem.Item, err error) {
+	c.switchLock.RLock()
+	defer c.switchLock.RUnlock()
+
+	if c.inactive {
+		return nil, ErrConnectionLost
+	}
+
+	val, err := c.rpcActor.Call(contract, method, args...)
+	if err != nil {
+		return nil, err
+	}
+
+	if val.State != HaltState {
```

Comment:
at the beginning of this PR i thought that i needed to just call one method instead of another so i wanted to copy behavior but now the rules have changed and now it seems useless so dropped

> I don't see a lot of value in notHaltStateError.

hm, btw, there also was `wrapNeoFSError` that used to be usefull but not for now so also dropped in a separate commit